src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/rca/hotshard/NodeShardKey.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.store.rca.hotshard;
+
+public class NodeShardKey {
+    private final String nodeId;
+    private final String shardId;
+
+    public NodeShardKey(String nodeId, String shardId) {
+        this.nodeId = nodeId;
+        this.shardId = shardId;
+    }
+
+    public String getNodeId() {
+        return this.nodeId;
+    }
+
+    public String getShardId() {
+        return this.shardId;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj != null && obj instanceof NodeShardKey) {
+            NodeShardKey key = (NodeShardKey)obj;
+            return nodeId.equals(key.nodeId) && shardId.equals(key.shardId);
+        }
+        return false;
+    }
+
+    @Override
+    public int hashCode() {
```

Comment:
We should probably use something like [this](https://github.com/opendistro-for-elasticsearch/performance-analyzer-rca/blob/f2dd3fdc106633fa9e59da345a9bddf843be6521/src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/response/NodeSummaryResponse.java#L70) for hashcode.

Reply:
Done, updated the hash code function.